Client.xcodeproj/project.pbxproj

```diff
@@ -13142,6 +13142,7 @@
 				MTL_ENABLE_DEBUG_INFO = INCLUDE_SOURCE;
 				MTL_FAST_MATH = YES;
 				ONLY_ACTIVE_ARCH = YES;
+				OTHER_SWIFT_FLAGS = "$(OTHER_SWIFT_FLAGS_common) -DMOZ_CHANNEL_FENNEC -DMOZ_TARGET_CREDENTIAL_PROVIDER";
```

Comment:
@lmarceau Let's move this into the XCConfig file if possible as this can easily get lost in the big settings change.
That would be the correct place. But let me know if there are any issues doing that and I'll jump on. 

Here is an example of a common file: https://github.com/mozilla-mobile/firefox-ios/blob/main/Client/Configuration/Common.xcconfig

Reply:
But Common.xcconfig will apply to all targets no? In my case I only want a flag to know this is running for the CredentialProvider target. Also, I used this method since NotificationService and ShareTo targets had the similar method, but with the flags `MOZ_TARGET_NOTIFICATIONSERVICE` & `MOZ_TARGET_SHARETO`

Reply:
Ah common.xcconfig is our top-level common file but we have others in the app as well I can show you tomorrow. I'd suggest treating `common.xcconfig` as an example, I have something I'd like to try with another config file. 

XCode's interface is very poor in terms of navigating through all these flags. Adding these custom flags inside the pbxproj means the information will get buried and we won't even be able to add an explanatory comment as to why we did it. With another xcconfig file, I am hoping we can avoid that or atleast try before we even add to XCode's pbxproj.

🤞🏼